src/Protocol/Smtp/ClientFactory.php

```diff
@@ -224,10 +224,14 @@ public static function fromString(string $dataSourceName):ClientFactory
                     $components['port'] ?? 25
                 );
                 break;
+            case 'null':
```

Comment:
I don't know if we should add `null://` to the client factory. A new `NullConnection` is not something that typically requires a factory. `new Client(new NullConnection())` is not harder than `ClientFactory::fromString()->newClient()`. Moreover, `NullConnection` is typically required during testing. So there is no argument in adding it for the developer's easiness. Unless, you have sound reasons to have it in the factory, I'd say we do not add it to the factory.

Reply:
> […] is not harder than […]

That is both fair and correct …

> Unless, you have sound reasons to have it in the factory, I'd say we do not add it to the factory.

My counter — that I am only adding for historical reference — is that my aversion to inconsistency is approximate to your aversion to setters :wink: 

However, personal ideology shouldn't stand in the way of a practical solution … so consider it "removed" :+1:

Reply:
Oh, and linking #10 as when that gets moving, I'll add your suggestion to the docs.